common/network-common/src/main/java/org/apache/spark/network/sasl/aes/AesCipher.java

```diff
@@ -0,0 +1,340 @@
+    this(new Properties(), configMessage.inKey, configMessage.outKey,
+      configMessage.inIv, configMessage.outIv);
+  }
+
+  /**
+   * Create AES crypto output stream
+   * @param ch The underlying channel to write out.
+   * @return Return output crypto stream for encryption.
+   * @throws IOException
+   */
+  public CryptoOutputStream CreateOutputStream(WritableByteChannel ch) throws IOException {
+    if (!outputStreamMap.containsKey(ch)) {
+      outputStreamMap.put(ch, new CryptoOutputStream(TRANSFORM, properties, ch, outKeySpec, outIvSpec));
+    }
+
+    return outputStreamMap.get(ch);
+  }
+
+  /**
+   * Create AES crypto input stream
+   * @param ch The underlying channel used to read data.
+   * @return Return input crypto stream for decryption.
+   * @throws IOException
+   */
+  public CryptoInputStream CreateInputStream(ReadableByteChannel ch) throws IOException {
+    if (!inputStreamMap.containsKey(ch)) {
+      inputStreamMap.put(ch, new CryptoInputStream(TRANSFORM, properties, ch, inKeySpec, inIvSpec));
+    }
+
+    return inputStreamMap.get(ch);
+  }
+
+  /**
+   * Add handlers to channel
+   * @param ch the channel for adding handlers
+   * @throws IOException
+   */
+  public void addToChannel(Channel ch) throws IOException {
+    ch.pipeline()
+      .addFirst(ENCRYPTION_HANDLER_NAME, new AesEncryptHandler(this))
+      .addFirst(DECRYPTION_HANDLER_NAME, new AesDecryptHandler(this));
+  }
+
+  /**
+   * Generate a request config message which send to remote peer.
+   * @param conf is the local transport configuration.
+   * @return Config message for sending.
+   */
+  public static AesConfigMessage requestConfigMessage(TransportConf conf) {
+    int keySize = conf.saslEncryptionAesCipherKeySizeBits();
+    if (keySize % 8 != 0) {
+      throw new IllegalArgumentException("The AES cipher key size in bits should be a multiple " +
+        "of byte");
+    }
+    return new AesConfigMessage(keySize/8, null, null, null, null);
+  }
+
+  /**
+   * Generate the configuration message according to request config message.
+   * @param configMessage The request config message comes from remote.
+   * @return Configuration message for sending.
+   */
+  public static AesConfigMessage responseConfigMessage(AesConfigMessage configMessage){
+
+    Properties properties = new Properties();
+    int keyLen = configMessage.keySize;
+
+    try {
+      int paramLen = CryptoCipherFactory.getCryptoCipher(AesCipher.TRANSFORM, properties)
+        .getBlockSize();
+      byte[] inKey = new byte[keyLen];
+      byte[] outKey = new byte[keyLen];
+      byte[] inIv = new byte[paramLen];
+      byte[] outIv = new byte[paramLen];
+
+      CryptoRandom random = CryptoRandomFactory.getCryptoRandom(properties);
+      random.nextBytes(inKey);
+      random.nextBytes(outKey);
+      random.nextBytes(inIv);
+      random.nextBytes(outIv);
+
+      configMessage.setParameters(keyLen, inKey, inIv, outKey, outIv);
```

Comment:
As with the previous comment, it's weird to modify input parameters. Instead, you should be returning a new message with the response data.

In fact, it feels like this method actually belongs in `SaslRpcHandler`, where it's called. There's nothing in this method that is related to the `AesCipher` class. It also needs a better name (e.g. `createResponseMessage`).